helpers/argumentParsers.go

```diff
@@ -2,622 +2,1374 @@ package helpers
+
+func ParseCloneFlags(rawValue uint64) (CloneFlagArgument, error) {
 	var f []string
-	if prot == 0x0 {
-		f = append(f, "PROT_NONE")
-	} else {
-		if prot&0x01 == 0x01 {
-			f = append(f, "PROT_READ")
-		}
-		if prot&0x02 == 0x02 {
-			f = append(f, "PROT_WRITE")
-		}
-		if prot&0x04 == 0x04 {
-			f = append(f, "PROT_EXEC")
-		}
+	if OptionAreContainedInArgument(rawValue, CLONE_VM) {
+		f = append(f, CLONE_VM.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_FS) {
+		f = append(f, CLONE_FS.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_FILES) {
+		f = append(f, CLONE_FILES.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_SIGHAND) {
+		f = append(f, CLONE_SIGHAND.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_PIDFD) {
+		f = append(f, CLONE_PIDFD.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_PTRACE) {
+		f = append(f, CLONE_PTRACE.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_VFORK) {
+		f = append(f, CLONE_VFORK.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_PARENT) {
+		f = append(f, CLONE_PARENT.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_THREAD) {
+		f = append(f, CLONE_THREAD.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_NEWNS) {
+		f = append(f, CLONE_NEWNS.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_SYSVSEM) {
+		f = append(f, CLONE_SYSVSEM.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_SETTLS) {
+		f = append(f, CLONE_SETTLS.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_PARENT_SETTID) {
+		f = append(f, CLONE_PARENT_SETTID.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_CHILD_CLEARTID) {
+		f = append(f, CLONE_CHILD_CLEARTID.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_DETACHED) {
+		f = append(f, CLONE_DETACHED.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_UNTRACED) {
+		f = append(f, CLONE_UNTRACED.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_CHILD_SETTID) {
+		f = append(f, CLONE_CHILD_SETTID.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_NEWCGROUP) {
+		f = append(f, CLONE_NEWCGROUP.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_NEWUTS) {
+		f = append(f, CLONE_NEWUTS.String())
 	}
+	if OptionAreContainedInArgument(rawValue, CLONE_NEWIPC) {
+		f = append(f, CLONE_NEWIPC.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_NEWUSER) {
+		f = append(f, CLONE_NEWUSER.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_NEWPID) {
+		f = append(f, CLONE_NEWPID.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_NEWNET) {
+		f = append(f, CLONE_NEWNET.String())
+	}
+	if OptionAreContainedInArgument(rawValue, CLONE_IO) {
+		f = append(f, CLONE_IO.String())
+	}
+	if len(f) == 0 {
+		return CloneFlagArgument{}, fmt.Errorf("no valid clone flag values present in raw value: 0x%x", rawValue)
```

Comment:
It might be that no flags were given as an argument (rawValue is 0), why do we want to return error in that case? Shouldn't we just return an empty string?
Same for all other options below.

Reply:
This is a good point, perhaps it should be 

`if len(f) == 0 && rawValue != 0 {`

Which would mean that it isn't an empty argument, but it isn't any valid one. WDYT?

Reply:
Another option is to check at the beginning of the function if `rawValue == 0 ` and return an empty string.
Then you can keep this test here